src/tracks/playabletrack/wavetrack/ui/WaveTrackControls.cpp

```diff
@@ -850,15 +791,16 @@ void WaveTrackMenuTable::SplitStereo(bool stereo)
    AudacityProject *const project = &mpData->project;
    auto channels = TrackList::Channels( pTrack );
 
-   // Unlink to make pan values independent, before changing them
-   TrackList::Get( *project ).UnlinkChannels( *pTrack );
-
    int totalHeight = 0;
    int nChannels = 0;
+
+   float pan = -1.0f;
    for (auto channel : channels) {
       auto &view = TrackView::Get( *channel );
-      if (stereo)
-         channel->SetPanFromChannelType();
+      if (stereo) {
+         channel->SetPan(pan);
```

Comment:
This is a bug : the second iteration will pan **both** channels hard right. This is because channels of one track now share the same `GainAndPan` sample track attachment.

Reply:
Right, I should unlink channels first

Reply:
Works now.